chanzuckerberg/idseq-web
# Add sample sort control to heatmap

# Description

See https://jira.czi.team/browse/IDSEQ-1259. This enables users to turn off clustering and sort by sample name. 

![image](https://user-images.githubusercontent.com/28797/63310740-b8161b80-c2b0-11e9-850b-97404fc281af.png)

![image](https://user-images.githubusercontent.com/28797/63310751-bf3d2980-c2b0-11e9-93c9-a2e79191edb3.png)

# Notes

* We are also going to add sorting taxa rows by genera. See https://jira.czi.team/browse/IDSEQ-1258 .
* We could add reverse alpha order if we want
* I left the current default to "cluster" though I think there is a good argument to make the default "alpha". 
* I adjusted the size of some of controls to make the new one fit in the row. I'm waiting on @happyimadesignr for a final visual design. 
* I think we should explain better what "clustering" does. I don't think the average user would guess that it is clustering by euclidean distance on the selected metric. 

# Tests

* Open a heatmap
* See default sort is "cluster" 
* Select "alpha"
* See heatmap change to correct order 
* Select "cluster"
* See heatmap return to previous order
